src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/Infrastructure/ServiceControlApi.cs

```diff
@@ -20,12 +23,62 @@ public bool CheckIsAvailable()
             return Get<object>("").Result != null;
         }
 
+        public async Task<FailedMessage> WaitForNewFailingMessages(string endpointName)
+        {
+            // NB: Not threadsafe! If some other process is interfering, this code is completely broken.
+            // Perhaps we should use a message mutator to hardcode the message id we're looking for instead?
+            var startMessages = await GetErrorsForEndpoint<IList<FailedMessage>>(endpointName).ConfigureAwait(false);
+            var startCount = startMessages.Count;
+
+            IList<FailedMessage> currentMessages;
+            int newCount;
+
+            do
+            {
+                Thread.Sleep(200);
+                currentMessages = await GetErrorsForEndpoint<IList<FailedMessage>>(endpointName).ConfigureAwait(false);
+                newCount = currentMessages.Count;
+            } while (newCount <= startCount);
+
+            var newFailedMessage = currentMessages.FirstOrDefault(m => !startMessages.Select(sm => sm.Id).Contains(m.Id));
```

Comment:
:+1: for grabbing the Id if we can. Not sure if we have an API to override the ID though.